src/steps/os/windows.rs

```diff
@@ -42,10 +42,20 @@ pub fn run_winget(ctx: &ExecutionContext) -> Result<()> {
 
     print_separator("winget");
 
-    ctx.run_type()
-        .execute(winget)
-        .args(["upgrade", "--all"])
-        .status_checked()
+    // check if silent install is enabled
+    if ctx.config().winget_silent_install() {
+        // execute with silent command
+        ctx.run_type()
+            .execute(winget)
+            .args(["upgrade", "--all", "--silent"])
+            .status_checked()
+    } else {
+        // execute without silent command
+        ctx.run_type()
+            .execute(winget)
+            .args(["upgrade", "--all"])
+            .status_checked()
+    }
```

Comment:
Nit: Let's do this to avoid a small amount of duplication:)

``` suggestion
    let mut args = vec!["upgrade", "--all"];
    if ctx.config().winget_silent_install() {
        args.push("--silent");
    }

    ctx.run_type().execute(winget).args(args).status_checked()
```

Reply:
The suggested code is not formatted, you may need to run `cargo fmt` yourself:D